remoto/process.py

```diff
@@ -37,6 +37,30 @@ def _remote_run(channel, cmd, **kw):
                     sys.stderr.flush()
 
         if process.poll() is not None:
+            # ensure we do not have anything pending in stdout or stderr
+            # unfortunately, we cannot abstract this repetitive loop into its
+            # own function because execnet does not allow for non-global (or
+            # even nested functions). This must be repeated here.
+            while True:
+                for descriptor in reads:
+                    if descriptor == process.stdout.fileno():
+                        read = process.stdout.readline()
+                        if read:
+                            channel.send({'debug': read})
+                            sys.stdout.flush()
+
+                    if descriptor == process.stderr.fileno():
+                        read = process.stderr.readline()
+                        if read:
+                            channel.send({'warning': read})
+                            sys.stderr.flush()
+                # At this point we have gone through all the possible
+                # descriptors and `read` was empty, so we now can break out of
+                # this since all stdout/stderr has been properly flushed to
+                # logging
+                if not read:
+                    break
```

Comment:
One question about this entire loop.  What would happen if stdout had a lot of data to read (say process.stdout.readline() returned some data, but not *all* the data), but stderr was None?

It seems like it would do a partial read on stdout, then do the read on stderr, returning None, then the for loop exits, read is None, and the whole thing exits...  Am I reading this wrong?

Seems like you might want to rearrange the logic to me be

```
for descriptor in...
     while True:
          if descriptor == stdout ....
               if read:
                    ....
               else:
                    break  # break out of while True, moving on to next descriptor....
           if descriptor == stderr ....